include/ignition/gazebo/EntityComponentManager.hh

```diff
@@ -487,6 +487,12 @@ namespace ignition
       /// \return True if there are any components with one-time changes.
       public: bool HasOneTimeComponentChanges() const;
 
+      /// \brief Get the components that are marked as periodic changes.
+      /// \return All the components that at least one entity marked as
+      /// periodic changes.
+      public: std::unordered_set<ComponentTypeId>
+          GetPeriodicChangedComponents() const;
```

Comment:
The function name is a bit misleading, because it's returning the types of the components, not the components.

Also, a small style nit is that we avoid using the `Get` prefix.

How about

```suggestion
          ComponentTypesWithPeriodicChanges() const;
```